content/blog/2019/06/2019-06-06-cloudfeature-gsoc-cp1.adoc

```diff
@@ -0,0 +1,103 @@
+
+=== Discoveries
+In order to make things clear, we need to understand the true nature of EWM. As we stated earlier, we want multiple agents to share disk spaces, and the main obstacle was they all have their own workspace directory (paths) by default. If we can modify the default workspace directory, then we might be able to utilize the shared NFS (Network File System).  And that is what EWM did. There are other features in like different strategies choosing a disk, but the way EWM to solve the problem is simple but effective.
+
+And by the way, from my perspective,the main problem, synchronization when visiting shared resources like two slave visit the shared file, is guaranteed by the implementation of the NFS itself, and the EWM didn't offer any synchronization mechanism to support concurrent execution. Jenkins itself is not designed to be synchronized. It is left to the users to partition their jobs and let the user manage syncronization. There might be some tricks to do that, but not within the scope of this blog.
+
+When we know what EWM did, we also realized : The Cloud feature is actually already supported. First, from the users perspective, they don't need to worry about the type of the disk, no matter it is a ext2 or ext3 or NFS or any other file system, they all look the same to host machine because of the concept of Virtual File System. If user can mount a disk to their host machine, then the EWM would be able to use them. So user can just create an EFS (Elastic File System, from AWS) in using whatever tools they like, and mount it. Then they can orchestra EFS with EWM.
+
+Then the problem becomes, should Jenkins be responsible to manage the NFS, for example, creating/managing/mounting/destroying the AWS EFS ? Well, my answer is not. There are some reasons. We will use AWS as example, but the same for other cloud providers.
+
+1. For small users, they don't need anybody to manage their jenkins clusters. AWS console gives them a user-friendly to manage the resources, and that would be enough.
+
+2. For medium/large users, it is impossible for them to manage Cloud resources manually (by console). They must use some resource management tools, like CloudFormation by AWS and Terraform by Hashicorp. And me personally are using these tools to manange clusters with thousands of AWS resources. From my perspective, I would like all the resources managed in one place, then that would make me easy to find the problem if something been created/destroyed unexpectedly. I won't let a second participation in my relationship with AWS, as they might potentially ruin everything. e.g, sometimes I had to manage resources using console, but this is even considered as a dangerous operation. Therefore, this plugin should not and not bother to manage any cloud resources. (There are plugins like ec2-plugin who do, but they are well managed, which is safe and used a lot).
+
```

Comment:
```suggestion
+
```
Another + is needed between any consecutive paragraphs and between a code block and paragraph.  See https://mrhaki.blogspot.com/2017/10/awesome-asciidoctor-using-paragraphs-in.html
Otherwise the lis numbering breaks (see CI build).

Reply:
@imaffe the same change is needed ~7 more times